test/e2e/vclient.go

```diff
@@ -143,13 +163,55 @@ func (c *vcClient) getSupervisorID() (string, error) {
 	}
 
 	for _, sv := range response.Items {
+		log.Info("Checking supervisor", "supervisor", sv.Info.Name, "status", sv.Info.ConfigStatus)
 		if sv.Info.ConfigStatus == "RUNNING" {
 			return sv.ID, nil
 		}
 	}
 	return "", fmt.Errorf("no valid supervisor found on vCenter")
 }
 
+func (c *vcClient) getStoragePolicyID() (string, error) {
+	urlPath := "/api/vcenter/storage/policies"
+	request, err := c.prepareRequest(http.MethodGet, urlPath, nil)
+	if err != nil {
+		return "", err
+	}
+	// response is a list of storage policy info
+	var response []storagePolicyInfo
+	if _, err = c.handleRequest(request, &response); err != nil {
+		return "", err
+	}
+
+	for _, po := range response {
+		log.Info("Checking storage policy", "policy", po.Name, "description", po.Description)
+		if strings.Contains(po.Name, "global") || strings.Contains(po.Name, "local") {
+			return po.Name, nil
+		}
+	}
+	return "", fmt.Errorf("no valid storage policy found on vCenter")
+}
+
+func (c *vcClient) getContentLibraryID() (string, error) {
```

Comment:
It looks you are trying to get the first content library? How do you ensure this can work for the tests? Could we filter with the test preferred resource to avoid unexpected failure?

BTW, could you add comments in the function?

Reply:
Getting the first content library should be OK.